src/main/java/org/jenkinsci/plugins/parallel_test_executor/ParallelTestExecutor.java

```diff
@@ -342,10 +343,18 @@ public String getDisplayName() {
         }
     }
 
-    static Predicate<FlowNode> stageNamePredicate(@Nonnull String stageName) {
+    static Predicate<FlowNode> stageNamePredicate(@Nonnull final String stageName) {
         // Arrays.asList in to prevent compiler warning due to unchecked generics/varargs whackiness.
         return Predicates.and(Arrays.asList(new NodeStepTypePredicate("stage"),
-                FlowScanningUtils.hasActionPredicate(LabelAction.class),
-                new NodeDisplayNamePredicate(stageName)));
+                new Predicate<FlowNode>() {
+                    @Override
+                    public boolean apply(@Nullable FlowNode input) {
+                        if (input != null) {
+                            LabelAction labelAction = input.getAction(LabelAction.class);
```

Comment:
🐜 Better to use getPersistentAction (performance micro-optimization!) but I think this is much better.